client/es/elasticsearch.go

```diff
@@ -87,17 +105,25 @@ type transportClient struct {
 func (c *transportClient) Perform(req *http.Request) (*http.Response, error) {
 	sp, err := c.startSpan(req)
 	if err != nil {
-		return nil, err
+		log.FromContext(req.Context()).Errorf("failed to start span: %v", err)
 	}
+	start := time.Now()
 	rsp, err := c.client.Perform(req)
 	if err != nil || rsp == nil {
 		trace.SpanError(sp)
 		return rsp, err
 	}
-	endSpan(sp, rsp)
+
+	observeResponse(req, sp, rsp, start)
```

Comment:
same as above ... maybe using defer would make this more consice

Reply:
generally I am avoiding defer's if the value they are capturing is generated inside the code block.
The reason is that the code is not really readable because you are capturing a variable that exists at a later stage which seems odd to read.